### DISCUSSION TITLE
FauxmoPhilipsLight_Basic.ino not detected by Amazon Echo

### DISCUSSION THREAD

**Discussion by dereknbartram** [MrGreensWorkshop/ArduinoPico_RasPiPicoW_AlexaEnabledSmartDevices#3] [Problems]
### Please check these items before go any further

- [X] I have watched [the related video](https://youtu.be/laeV6GgGJMA) on YouTube.
- [X] I have searched the existing [Issues](https://github.com/MrGreensWorkshop/arduinoPico_RasPiPicoW_AlexaEnabledSmartDevices/issues/).
- [X] I have searched the existing [Discussions](https://github.com/MrGreensWorkshop/arduinoPico_RasPiPicoW_AlexaEnabledSmartDevices/discussions/).
- [X] I have checked the existing Issues in the [fauxmoESP repo](https://github.com/vintlabs/fauxmoESP/issues/) and the answer was not there.
- [X] I have checked the existing Issues in the [arduino-pico repo](https://github.com/earlephilhower/arduino-pico/issues/) and the answer was not there.
- [X] I have checked the existing Issues in the [AsyncTCP_RP2040W repo ](https://github.com/khoih-prog/AsyncTCP_RP2040W/issues/) and the answer was not there.

### Are you sure this is a bug?

- [X] I'm sure this is definitely a bug.

### Describe the bug

Compiled and deployed FauxmoPhilipsLight_Basic.ino example to a bare PI PICO W, however Alexa cannot discover it

### Expected Behavior

New device(s) detected

### Steps To Reproduce

Compiled and deployed FauxmoPhilipsLight_Basic.ino to a bare PI PICO W
Confirmed device on network by seeing expected response from http://PI_IP_ADDRESS/api/user/lights
Searched for new devices both from device and Alexa app

### Which device have you used?

- [X] Raspberry Pi Pico W
- [ ] ESP8266
- [ ] ESP32 (This project hasn't been tested on ESP32)

**Comment by MrGreensWorkshop** [comment]
- What version did you use for the Arduino Pico and AsyncTCP_RP2040W?
- What is your Amazon Echo device generation?
- Did you use [FauxmoPhilipsLight_Basic.ino](https://github.com/MrGreensWorkshop/ArduinoPico_RasPiPicoW_AlexaEnabledSmartDevices/tree/main/examples/FauxmoPhilipsLight_Basic) with [FauxmoPhilipsLight](https://github.com/MrGreensWorkshop/ArduinoPico_RasPiPicoW_AlexaEnabledSmartDevices/tree/main/library/FauxmoPhilipsLight)?

**Comment by dereknbartram** [comment]
Hi, thanks for the message.

'First Release' for the examples code
PI PICO W, and arduino-pico version 3.0.0
AsyncTCP_RP2040W = 1.3.0
Echo Dot, generation 3
Yes: [FauxmoPhilipsLight_Basic.ino](https://github.com/MrGreensWorkshop/ArduinoPico_RasPiPicoW_AlexaEnabledSmartDevices/tree/main/examples/FauxmoPhilipsLight_Basic) with [FauxmoPhilipsLight](https://github.com/MrGreensWorkshop/ArduinoPico_RasPiPicoW_AlexaEnabledSmartDevices/tree/main/library/FauxmoPhilipsLight)?

**Comment by MrGreensWorkshop** [comment]
Hi,
No problem, I hope you can figure it out.

As I explained in the Readme file, these are the versions that I tested.

- Arduino Pico [v2.6.0](https://github.com/earlephilhower/arduino-pico/tree/2.6.0)
- AsyncTCP_RP2040W [v1.1.0](https://github.com/khoih-prog/AsyncTCP_RP2040W/tree/v1.1.0)
- gen 1 and gen 2 devices.

I also mentioned that, since I only have first and second generation Amazon Echo devices, I didn't test with third generation devices. 

All I can say is to try the same versions that I tested.

**Comment by dereknbartram** [comment]
Understood, will give that a try.
When you were writing this, did you have any issues with state change commands being sent, your code responding to the change with success, the Alexa app showing a state change, and then shortly after showing "Device is unresponsive"? I'm writing a micropython port and can't get state changes to 'stay'

**Comment by MrGreensWorkshop** [comment]
> When you were writing this, did you have any issues with state change commands being sent, your code responding to the change with success, the Alexa app showing a state change, and then shortly after showing "Device is unresponsive"?

When using the Alexa app, please give it a little time. Sometimes exiting the menu and opening the device again helps refresh its state. As I demonstrate in the [YouTube video](https://www.youtube.com/watch?v=laeV6GgGJMA&t=270s), reading the status of lights works as expected.

**Comment by JanikB310** [comment]
**I made it work with an Echo 3th gerneration.** 
First, I had only an sonos speaker with built in alexa and it didnt work. Then i inclueded an echo 3th generation and it worked.
